addons/website/static/src/snippets/s_popup/options.js

```diff
@@ -92,16 +92,19 @@ options.registry.SnippetPopup = options.Class.extend({
     //--------------------------------------------------------------------------
 
     /**
-     * Moves the snippet in footer to be common to all pages
-     * or inside wrap to be on one page only
+     * Moves the snippet in #o_shared_blocks to be common to all pages or inside
+     * the first editable oe_structure in the main to be on current page only.
      *
      * @see this.selectClass for parameters
      */
     moveBlock: function (previewMode, widgetValue, params) {
-        const containerEl = this.$target[0].ownerDocument.querySelector(widgetValue === 'moveToFooter' ? 'footer' : 'main');
-        const whereEl = $(containerEl).find('.oe_structure:o_editable')[0];
+        const selector = widgetValue === 'allPages' ?
+            '#o_shared_blocks' : 'main .oe_structure:o_editable';
+        const whereEl = $(this.$target[0].ownerDocument).find(selector)[0];
         const popupEl = this.$target[0].closest('.s_popup');
         whereEl.prepend(popupEl);
+        this.$target[0].ownerDocument.querySelector('#o_shared_blocks')
+            .dispatchEvent(new Event('content_changed'));
```

Comment:
Could be nice to have comment here. I don't understand why we would need to trigger this on o_shared_blocks only. I don't understand why it was not needed before (when we used the footer) either?

Reply:
Indeed, it was required with the public widget solution but it's useless now, I remove it :+1: